src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/DefaultRazorFormattingService.cs

```diff
@@ -90,15 +90,25 @@ internal class DefaultRazorFormattingService : RazorFormattingService
             return filteredEdits;
         }
 
-        public override async Task<TextEdit[]> ApplyFormattedEditsAsync(
+        public override Task<TextEdit[]> FormatOnTypeAsync(DocumentUri uri, DocumentSnapshot documentSnapshot, RazorLanguageKind kind, TextEdit[] formattedEdits, FormattingOptions options, CancellationToken cancellationToken)
+            => ApplyFormattedEditsAsync(uri, documentSnapshot, kind, formattedEdits, options, bypassValidationPasses: false, collapseEdits: false, automaticallyAddUsings: false, cancellationToken: cancellationToken);
+
+        public override Task<TextEdit[]> FormatCodeActionAsync(DocumentUri uri, DocumentSnapshot documentSnapshot, RazorLanguageKind kind, TextEdit[] formattedEdits, FormattingOptions options, CancellationToken cancellationToken)
+            => ApplyFormattedEditsAsync(uri, documentSnapshot, kind, formattedEdits, options, bypassValidationPasses: true, collapseEdits: false, automaticallyAddUsings: true, cancellationToken: cancellationToken);
+
+        public override Task<TextEdit[]> FormatSnippetAsync(DocumentUri uri, DocumentSnapshot documentSnapshot, RazorLanguageKind kind, TextEdit[] formattedEdits, FormattingOptions options, CancellationToken cancellationToken)
+            => ApplyFormattedEditsAsync(uri, documentSnapshot, kind, formattedEdits, options, bypassValidationPasses: true, collapseEdits: true, automaticallyAddUsings: false, cancellationToken: cancellationToken);
+
+        private async Task<TextEdit[]> ApplyFormattedEditsAsync(
             DocumentUri uri,
             DocumentSnapshot documentSnapshot,
             RazorLanguageKind kind,
             TextEdit[] formattedEdits,
             FormattingOptions options,
-            CancellationToken cancellationToken,
-            bool bypassValidationPasses = false,
-            bool collapseEdits = false)
+            bool bypassValidationPasses,
+            bool collapseEdits,
+            bool automaticallyAddUsings,
```

Comment:
We may want to build or re-use our options object eventually to avoid making this method clunky with loads of parameters. I'm sure there'll e a critical mass when that'll be preferable at least 😆